homeassistant/components/bluetooth/models.py

```diff
@@ -283,14 +284,21 @@ def set_disconnected_callback(
 
     async def connect(self, **kwargs: Any) -> bool:
         """Connect to the specified GATT server."""
-        if not self._backend:
+        if (
+            not self._backend
+            or not self.__ble_device
+            or not self._async_get_backend_for_ble_device(self.__ble_device)
+        ):
             assert MANAGER is not None
             wrapped_backend = (
                 self._async_get_backend() or self._async_get_fallback_backend()
             )
-            self._backend = wrapped_backend.client(
+            self.__ble_device = (
                 await freshen_ble_device(wrapped_backend.device)
-                or wrapped_backend.device,
+                or wrapped_backend.device
```

Comment:
Since we now always reuse the `self.__ble_device` object, should we ever set it to `None` ?

Reply:
It shouldn't ever be none since the address can't change without constructing a new client, and if we can't connect we are going to raise